lib/resque/job.rb

```diff
@@ -276,6 +319,9 @@ def after_hooks
 
     def failure_hooks
       @failure_hooks ||= Plugin.failure_hooks(payload_class)
+    rescue NameError
+      # if the payload class doesn't exist, there are no hooks
```

Comment:
This addresses a bug seen in production when old jobs are handled after their code has been deleted.

This could be PRed to master.